src/main/java/redis/clients/jedis/JedisClientSideCache.java

```diff
@@ -0,0 +1,45 @@
+package redis.clients.jedis;
+
+import redis.clients.jedis.exceptions.JedisException;
+
+public class JedisClientSideCache extends Jedis {
+
+  private final ClientSideCache cache;
+
+  public JedisClientSideCache(final HostAndPort hostPort, final JedisClientConfig config) {
```

Comment:
How about to use Proxy pattern here? Instead of introducing new connection type you can introduce a proxy object that decides whether you need to read against cache or delegate it to another "real" connection against Redis.

Aggregation is always preferable against inheritance :)

Reply:
I've found it easier to **start**.